File: CsEquivalents/ModuleExample.cs
using System;
using System.Collections;
using Microsoft.FSharp.Core;

using CsEquivalents.RecordTypeExamples;

namespace CsSample
{

    public static class ModuleExample
    {
        /// <summary>
        ///  add two numbers
        /// </summary>
        public static int Add(int x, int y)
        {
            return x + y;
        }

        /// <summary>
        ///  add 1 to a number
        /// </summary>
        public static int Add1(int x)
        {
            return x + 1;
        }

        [Serializable]
        public class Something
        {
            public Something()                
            {
            }
        }

        /// <summary>
        ///  Create a submodule
        /// </summary>
        public static class GameFunctions
        {
            /// <summary>
            /// Create a game with score=12
            /// </summary>
            public static FinalGameScore CreateGame(string name)
            {
                return new FinalGameScore(name, 12);
            }

            /// <summary>
            /// Change the score for an existing game
            /// </summary>
            public static FinalGameScore ChangeScore(int newScore, FinalGameScore game)
            {
                return new FinalGameScore(game.Game, newScore);
            }

            /// <summary>
            ///  Example of a higher order function
            /// </summary>
            public static FinalGameScore MapScore(FSharpFunc<int, int> f, FinalGameScore game)
            {
                return new FinalGameScore(game.Game, f.Invoke(game.FinalScore));
            }
        }



    }
}

File: FsExamples/ModuleExample.fs
module ModuleExample


/// add two numbers
let Add x y = x + y

/// add 1 to a number
let Add1 x = x + 1


/// Create a submodule
module GameFunctions =
    open RecordTypeExamples

    /// Create a game with score=12
    let CreateGame name = {Game=name; FinalScore=12}

    /// Change the score for an existing game
    let ChangeScore newScore game = 
        {game with FinalScore=newScore}

    /// Example of a higher order function
    let MapScore f game = 
        {game with FinalScore=f game.FinalScore}
